khal/ui/__init__.py

```diff
@@ -904,23 +907,18 @@ def keypress(self, size, key):
 
         rval = super().keypress(size, key)
         if self.focus_event:
-            if key in self._conf['keybindings']['view'] or \
+            if key in self._conf['keybindings']['view'] and \
+                    prev_shown == self.focus_event.recuid:
+                # the event in focus is already viewed -> edit
+                if self.delete_status(self.focus_event.recuid):
+                    self.pane.window.alert(('light red', 'This event is marked as deleted'))
```

Comment:
Of course you're right, this slipped through when going from latest release to current master 😳